controllers/common.go

```diff
@@ -17,21 +17,51 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+// LeaseTruncatedError indicates that the requested lease renewal duration is
+// less than expected
+type LeaseTruncatedError struct {
+	Expected int
+	Actual   int
+}
+
+func (l *LeaseTruncatedError) Error() string {
+	return fmt.Sprintf("lease renewal duration was truncated from %ds to %ds",
+		l.Expected, l.Actual)
+}
+
```

Comment:
Suggest declaring that `LeaseTruncatedError` must implement `error`

```suggestion
var _ error = (*LeaseTruncatedError)(nil)
```